### ISSUE TITLE
[New Feature]: Deploy EKS to MCP venues with Terraform

### ISSUE THREAD

**Issue by LucaCinquini** [unity-sds/unity-sps-prototype#255]
Start from the current CS Terraform scripts to instantiate a completely new EKS cluster onto a target CP venue.
Plan for the TF scripts to be customizable with SPS specific add-ons such as Karpenter (for autoscaling of Kubernetes nodes).
Make sure that the scripts also deploy the SPS ELB that is connects to the CS API Gateway.

Acceptance Criteria:
o Demonstrated deployment of EKS onto mcp-venue-dev and mcp-venue-test
o CI/CD pipeline for automatic EKS deployment"

**Comment by mike-gangl** [comment]
Can you provide an update on this ticket?

**Comment by LucaCinquini** [comment]
Ryan was successful in modifying the CS EKS script to add sone functionality:
o The ELB controller
o Extended Kubernetes permissions
o A placeholder to add SPS specific add-ons to the Kubernetes cluster

Issued a PR to CS and now finalizing a script that can be executed to trigger Terraform and start deploy EKS on a target venue.

**Comment by LucaCinquini** [comment]
PR to unity-cs-infra was merged: https://github.com/unity-sds/unity-cs-infra/pull/42/files

Luca would like to try the deployment script before closing the ticket.

**Comment by ryanhunter-jpl** [comment]
Merged in changes for SPS as well - https://github.com/unity-sds/unity-sps-prototype/pull/271.

To deploy a cluster, from the root of the repository run - 
`cd terraform-unity/modules/terraform-eks-cluster`
`terraform init`
`terraform apply -var "cluster_name=my-cluster-name-here"`

For testing purposes (i.e. my development cycle) run - 
`cd terraform-unity/modules/terraform-eks-cluster`
`./validate.sh`

The `validate.sh` script does the following:
1. generates a random alpha-numeric cluster name
2. creates a new terraform workspace with the cluster name
3. runs `terraform init` in the workspace
4. runs `terraform apply -var "cluster_name=random-cluster-name"`
5. initializes a kubeconfig for the cluster
6. verifies that the cluster is reachable with `kubectl get all -A`
7. teardown the cluster with `terraform destroy`

I found it to be useful automation to verify changes weren't breaking everything so I included it in the PR.
